### ISSUE TITLE
corrosive skin viper + slark dark pact

### ISSUE THREAD

**Issue by Artur11bobruisk** [ValveSoftware/Dota2-Gameplay#6256]
### Ability name

corrosive skin

### Description

vipers corrosive skin  with aghanim scepter deals incredible damage to the slark when it uses dark pact. I test this in lobby. others skills like diabolic edict and other abilities that deal damage at intervals of less than a second. Under their influence, corrosive skin deals damage with an interval of 1 second. But only dark pact deals corrosive skin damage by slark for each 'portion' (in 0.1 second interval) of dark pact damage

### Example Match ID (and possibly Timestamp)

all pick 28.12.2022 20:49 GMT+3  match number 6941380626.   examples on 19:46, 21:22

**Comment by irismus** [comment]
Viper's aghs changes how Corrosive Skin works. Keep in mind that Corrosive Skin debuff is not dispellable (i.e. only dispel on death).

When Dark Pact is active, it removes the buff in 0.1-second intervals.
Corrosive Skin with aghs now applies the DPS and debuff in the same intervals, and since it is not dispellable, it therefore damages Slark each time Dark Pact dispels the buff because the debuff is being reapplied.

Diabolic Edict does not use debuff to deal damage.

**Comment by bezhdeyire** [comment]
This is dont should work like this. Dark Pact purpose is to remove debuffs, not to kill you.

**Comment by shaileshgarg** [comment]
Viper Corrosive Skin applies to enemy attacking Viper. Slark dark pact hitting viper counts as attack. But is does dispels slark. So dark pact dispels slark, but it hit viper so he again gets the debuff. This happens every 0.1s of Slarks darkpact.

Viper Corrosive Skin without Sceptor does damage 1 sec after it is applied, so slark Dark pacting on viper will refresh Corrosive Skin every 0.1s, can be checked in Demo. But there is no damage as damage starts 1 sec after Corrosive Skin is applied.


Viper Corrosive Skin with Sceptor does damage instantly after it is applied, so slark Dark pacting on viper will refresh Corrosive Skin every 0.1s, and receive the instant damage tick every 0.1s.

This is working as intented.

**Comment by bezhdeyire** [comment]
Windranger and Skymage also have the same mechanic of skills, but dont take damage like slark.

**Comment by cadaeic** [comment]
This is not a bug per se but still an exploit, since it's definitely not what is intended. No repeating debuff that can be purged should act this way. There's no reason purchasing the scepter on Viper changes this one particular interaction.

Slark doesn't kill himself by using Dark Pact under Chaos Meteor, for comparison. And Macropyre was changed to be undispellable. (Mars shard has the same bug though.)

**Comment by shaileshgarg** [comment]
> This is not a bug per se but still an exploit, since it's definitely not what is intended. No repeating debuff that can be purged should act this way. There's no reason purchasing the scepter on Viper changes this one particular interaction.
> 
> Slark doesn't kill himself by using Dark Pact under Chaos Meteor, for comparison. And Macropyre was changed to be undispellable. (Mars shard has the same bug though.)

Yes, as the Burn Damage of Chaos Meteor is appplied after 1s of the Buff being applied.

Tested with Mars Shard that doesn't seem to have the issue. It deals damage after 0.5s of the buff being applied, so no issue like Viper Aghanim

**Comment by cadaeic** [comment]
> Tested with Mars Shard that doesn't seem to have the issue. It deals damage after 0.5s of the buff being applied, so no issue like Viper Aghanim

Tested in a lobby, the aoe effect applies the damage on the first tick of the debuff. In this case the debuff is being reapplied on a 0.3s interval, so the effect isn't as dramatic, but still makes Slark take about 2x damage from the burning trail.
